packages/@react-aria/color/src/useColorWheel.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {ColorWheelAriaProps, ColorWheelAriaResult, ColorWheelState} from '@react-types/color';
+import {focusWithoutScrolling, mergeProps} from '@react-aria/utils';
+import React, {useCallback, useRef} from 'react';
+import {useMove} from '@react-aria/interactions';
+
+function degToRad(deg: number) {
+  return deg * Math.PI / 180;
+}
+
+function radToDeg(rad: number) {
+  return rad * 180 / Math.PI;
+}
+
+// 0deg = 3 o'clock. increses clockwise
+function angleToCartesian(angle: number, radius: number): {x: number, y: number} {
+  let rad = degToRad(360 - angle + 90);
+  let x = Math.sin(rad) * (radius);
+  let y = Math.cos(rad) * (radius);
+  return {x, y};
+}
+function cartesianToAngle(x: number, y: number, radius: number): number {
+  let deg = radToDeg(Math.atan2(y / radius, x / radius));
+  return (deg + 360) % 360;
+}
+
+export function useColorWheel(props: ColorWheelAriaProps, state: ColorWheelState): ColorWheelAriaResult {
+  let {inputRef, containerRef, isDisabled, step = 1, innerRadius, outerRadius} = props;
+
+  let thumbRadius = (innerRadius + outerRadius) / 2;
+
+  let focusInput = useCallback(() => {
+    if (inputRef.current) {
+      focusWithoutScrolling(inputRef.current);
+    }
+  }, [inputRef]);
+
+  let stateRef = useRef<ColorWheelState>(null);
+  stateRef.current = state;
```

Comment:
I tried this using using a ref in `useMove` for these handler callbacks. But it still doesn't work as you'd expect:
```js
      let [state, setState] = useState([]);

      let moveProps = useMove({
        onMoveStart(e) {
          setState([...state, e.type]);
        },
        onMove(e) {
          setState([...state, e.type]);
        },
        onMoveEnd(e) {
          setState([...state, e.type]);
        }
      });
```
But after `touchMove`, `state === ["move"]` which is wrong, this is because`onMoveStart` and `onMove` were invoked in the same "react tick", so the state in `onMove` is stale anyway).

Can this even be solved in useMove to make it transparent to the hook consumer?

Reply:
Does it work if you use the `setState` function syntax? `setState(state => [...state, e.type])`